### PR TITLE
[PP-49] Add handlers to create/delete/vote tasks

### PR DESCRIPTION
**Pull request by andreikaptsiuh**
Task: https://andrei-lysiuk.atlassian.net/browse/PP-49
Description: 
- Add handlers to create/delete/vote tasks
image
image

### CODE REVIEW

File: src/components/Task/Task.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  return (
+    <div className={classNames(
+      styles.task, {
+        [styles.task__active]: task.isActive === true,
+      },
+    )}
+    >
+
+      <div className={styles.task__head}>
+        <h5>{`PP-${task.id}`}</h5>
```
Comment (mrakofff) [inline src/components/Task/Task.tsx]: why `pp-` prefix?

File: src/components/Task/Task.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        <Button
+          className={styles.remove__btn}
+          onClick={removeTask}
```
Comment (mrakofff) [inline src/components/Task/Task.tsx:55]: should we add confirm to exclude random click?

File: src/components/Task/Task.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      </div>
+      <p className={styles.task__title}>{task.title}</p>
+      <div className={styles.task__bottom}>
+        <PrimaryButton
+          className={styles.vote_btn}
+          onClick={setActiveTask}
+        >
+          Voting now...
+        </PrimaryButton>
+        <p>{task.score}</p>
```
Comment (mrakofff) [inline src/components/Task/Task.tsx]: Admin should be able to set task score after all users vote to save task score in db

File: src/components/CreateTaskPanel/CreateTaskPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  const createTask = () => {
+    if (taskTitle !== '') {
+      const task = {
+        title: taskTitle,
+        roomId,
+      };
+
+      socketService.emit(SocketEvent.TaskCreate, task);
+      taskEditorHandler();
```
Comment (mrakofff) [inline src/components/CreateTaskPanel/CreateTaskPanel.tsx:32]: you will get an error coz event is undefined inside handler : `const { value } = event.target;`

**Review by mrakofff — CHANGES_REQUESTED** [review]

File: src/components/Task/Task.tsx (hunk 1 above)
Comment (andreikaptsiuh) [reply src/components/Task/Task.tsx]: fix

File: src/components/CreateTaskPanel/CreateTaskPanel.tsx (hunk 4 above)
Comment (andreikaptsiuh) [reply src/components/CreateTaskPanel/CreateTaskPanel.tsx:32]: fixed

File: src/components/Task/Task.tsx (hunk 2 above)
Comment (andreikaptsiuh) [reply src/components/Task/Task.tsx:55]: fix

File: src/components/Task/Task.tsx (hunk 3 above)
Comment (andreikaptsiuh) [reply src/components/Task/Task.tsx]: fix

File: src/models/SendTaskModel/SendTaskModel.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+interface task {
```
Comment (mrakofff) [inline src/models/SendTaskModel/SendTaskModel.ts]: Should it be `INewTask`?

**Review by mrakofff — CHANGES_REQUESTED** [review]

File: src/models/SendTaskModel/SendTaskModel.ts (hunk 5 above)
Comment (andreikaptsiuh) [reply src/models/SendTaskModel/SendTaskModel.ts]: ok

File: src/components/Task/Task.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+  const [score, setScore] = useState(task.score);
+
+  const setTaskScore: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const newScore = e.target.value;
+    setScore(newScore);
+  };
+
+  const updateTaskScore: ChangeEventHandler<HTMLInputElement> = () => {
+    socketService.emit(SocketEvent.TaskUpdateScore, {
```
Comment (mrakofff) [inline src/components/Task/Task.tsx]: emit only if `task.score !== score`

File: src/components/Task/Task.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+  const removeTask = () => {
+    // eslint-disable-next-line no-restricted-globals
+    const res = confirm('Are you want to delete this task?');
+    if (res) {
+      socketService.emit(SocketEvent.TaskDelete, task.id);
+    }
+  };
+
+  const [score, setScore] = useState(task.score);
```
Comment (mrakofff) [inline src/components/Task/Task.tsx]: mote to the top of component

**Review by mrakofff — APPROVED** [review]

File: src/components/Task/Task.tsx (hunk 7 above)
Comment (andreikaptsiuh) [reply src/components/Task/Task.tsx]: fix

File: src/components/Task/Task.tsx (hunk 6 above)
Comment (andreikaptsiuh) [reply src/components/Task/Task.tsx]: fix

**Review by mrakofff — APPROVED** [review]

**Review by MakFill — APPROVED** [review]

File: src/store/actions/taskActions.ts [hunk 8]
```diff
@@ -10,15 +10,17 @@ export const taskActions = {
...
+          .map((task) => new TaskModel(task))
+          .sort((taskOne, taskTwo) => taskOne.id - taskTwo.id);
         return tasks;
       } catch (err) {
         return rejectWithValue(err);
       }
     }),
   addTask: createAction<ITask>('[TASKS]:addTask'),
-  deleteTask: createAction<ITask['id']>('[TASKS]:deleteTask'),
+  deleteTask: createAction<number>('[TASKS]:deleteTask'),
```
Comment (mrakofff) [inline src/store/actions/taskActions.ts]: any reason for update?

File: src/store/actions/taskActions.ts [hunk 9]
```diff
@@ -10,15 +10,17 @@ export const taskActions = {
     async (_, { rejectWithValue, getState }) => {
       try {
         const roomId = roomStateSelectors.roomId(getState() as RootState);
-        const tasksFromBE: ITaskFromBE[] = await httpClient.http.get(`/api/task?roomId=${roomId}`);
-        const tasks = tasksFromBE.map((task) => new TaskModel(task));
+        const tasksFromBE: ITaskFromBE[] = await httpClient.http.get(`/api/tasks?roomId=${roomId}`);
+        const tasks = tasksFromBE
+          .map((task) => new TaskModel(task))
+          .sort((taskOne, taskTwo) => taskOne.id - taskTwo.id);
```
Comment (mrakofff) [inline src/store/actions/taskActions.ts:16]: should we sort by task date or even sort or it should be done by `BE`?

File: src/components/Task/Task.tsx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+import { socketService } from '@/services';
+import { SocketEvent } from '@/utils/enums';
+import { cards } from '@/mocks/cards';
+
+interface ITaskProps {
+  task: ITask;
+}
+
+export const Task: React.FC<ITaskProps> = ({ task }) => {
+  const defaultScore = task.score === '0' ? '?' : task.score;
```
Comment (mrakofff) [inline src/components/Task/Task.tsx]: perhaps you should not set `0` in model and leave `null` if you do this check. Also move it to `useState` callback https://ru.reactjs.org/docs/hooks-reference.html#lazy-initial-state

File: src/components/Task/Task.tsx [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+        id: task.id,
+        score: newScore,
+      });
+    }
+  };
+
+  return (
+    <div className={classNames(
+      styles.task, {
+        [styles.task__active]: task.isActive === true,
```
Comment (mrakofff) [inline src/components/Task/Task.tsx]: could be simplified with `[styles.task__active]: task.isActive`

File: src/components/Issues/Issues.tsx [hunk 12]
```diff
@@ -36,8 +38,8 @@ export const Issues: React.FC = () => {
     dispatch(taskActions.addTask(newTask));
   };
 
-  const deleteTaskHandler = (task: ITaskFromBE) => {
-    dispatch(taskActions.deleteTask(task.id));
+  const deleteTaskHandler = (id: number) => {
```
Comment (mrakofff) [inline src/components/Issues/Issues.tsx]: use type from Task interface

**Review by mrakofff — CHANGES_REQUESTED** [review]

File: src/components/Task/Task.tsx (hunk 11 above)
Comment (andreikaptsiuh) [reply src/components/Task/Task.tsx]: fix

File: src/components/Task/Task.tsx (hunk 10 above)
Comment (andreikaptsiuh) [reply src/components/Task/Task.tsx]: fix

File: src/store/actions/taskActions.ts (hunk 9 above)
Comment (MakFill) [reply src/store/actions/taskActions.ts:16]: we should sort by id - it's similar like sort by date

File: src/components/Issues/Issues.tsx (hunk 12 above)
Comment (MakFill) [reply src/components/Issues/Issues.tsx]: refactored

File: src/store/actions/taskActions.ts (hunk 8 above)
Comment (MakFill) [reply src/store/actions/taskActions.ts]: refactored

**Pull request merged**